etl/steps/data/garden/urbanization/2024-10-14/ghsl_degree_of_urbanisation.py

```diff
@@ -0,0 +1,126 @@
+
+# Regions for which aggregates will be created.
+REGIONS = [
+    "North America",
+    "South America",
+    "Europe",
+    "Africa",
+    "Asia",
+    "Oceania",
+    "Low-income countries",
+    "Upper-middle-income countries",
+    "Lower-middle-income countries",
+    "High-income countries",
+]
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("ghsl_degree_of_urbanisation")
+    # Load regions dataset.
+    ds_regions = paths.load_dataset("regions")
+
+    # Load income groups dataset.
+    ds_income_groups = paths.load_dataset("income_groups")
+    # Read table from meadow dataset.
+    tb = ds_meadow["ghsl_degree_of_urbanisation"].reset_index()
+
+    #
+    # Process data.
+    #
+    tb = geo.harmonize_countries(
+        df=tb, countries_file=paths.country_mapping_path, excluded_countries_file=paths.excluded_countries_path
+    )
+
+    # Pivot the table to each indicator as a column.
+    tb = tb.pivot(index=["country", "year"], columns="indicator", values="value").reset_index()
+    columns_to_aggregate = [
+        "rural_total_population",
+        "urban_centre_population",
+        "urban_cluster_population",
+        "rural_total_area",
+        "urban_centre_area",
+        "urban_cluster_area",
+    ]
+
+    aggr_dict = {col: "sum" for col in columns_to_aggregate}
+    # Add region aggregates.
+    tb = geo.add_regions_to_table(
+        tb,
+        aggregations=aggr_dict,
+        regions=REGIONS,
+        ds_regions=ds_regions,
+        ds_income_groups=ds_income_groups,
+        min_num_values_per_year=1,
+    )
+
+    # Calculate "share of" indicators
+    tb["total_population"] = tb[["rural_total_population", "urban_centre_population", "urban_cluster_population"]].sum(
+        axis=1
+    )
+    tb["total_area"] = tb[["rural_total_area", "urban_centre_area", "urban_cluster_area"]].sum(axis=1)
+
+    tb["urban_total_popshare"] = (
+        (tb["urban_centre_population"] + tb["urban_cluster_population"]) / tb["total_population"]
+    ) * 100
+
+    tb["urban_total_share"] = (tb["urban_centre_area"] + tb["urban_cluster_area"]) / tb["total_area"] * 100
+
+    tb["rural_total_share"] = tb["rural_total_area"] / tb["total_area"] * 100
+    tb["rural_total_popshare"] = tb["rural_total_population"] / tb["total_population"] * 100
+
+    tb["urban_cluster_popshare"] = tb["urban_cluster_population"] / tb["total_population"] * 100
+    tb["urban_centre_popshare"] = tb["urban_centre_population"] / tb["total_population"] * 100
+
+    tb["urban_cluster_share"] = tb["urban_cluster_area"] / tb["total_area"] * 100
+    tb["urban_centre_share"] = tb["urban_centre_area"] / tb["total_area"] * 100
+
+    tb = tb.drop(columns=["total_population", "total_area"])
+
+    # Create two new dataframes to separate data into estimates and projections (pre-2025 and post-2025 (five year intervals)))
+    past_estimates = tb[tb["year"] < 2025].copy()
```

Comment:
Just as a visual preference, you can maybe keep 2020 with data for projections, so the dotted line in [this chart](http://staging-site-urbanization-oct/grapher/urban-share-european-commission) can start where the estimated finished